version: '3.8'

services:
  django:
    build: .
    container_name: django-app
    restart: unless-stopped
    ports:
      - "3015:3015"
    env_file:
      - .env
    depends_on:
      - redis-django
    command: python manage.py runserver 0.0.0.0:3015
    networks:
      - django-net

  celery-worker:
    build: .
    container_name: celery-worker
    restart: unless-stopped
    depends_on:
      - redis-django
      - django
    command: celery -A descriptive_analysis worker --loglevel=info --pool=solo
    env_file:
      - .env
    networks:
      - django-net

  redis-django:
    image: redis:6.2-alpine
    container_name: redis-django
    restart: unless-stopped
    ports:
      - "6381:6379"
    volumes:
      - ./evaluation/redis.conf:/usr/local/etc/redis/redis.conf
    command: [ "redis-server", "/usr/local/etc/redis/redis.conf" ]
    networks:
      - django-net

networks:
  django-net:
    name: django-net
